src/drivers/driver_zephyr.c

```diff
@@ -1812,7 +1823,11 @@ static int wpa_drv_zep_set_supp_port(void *priv,
 
 #ifdef CONFIG_NET_DHCPV4
 	if (authorized) {
-		net_dhcpv4_restart(iface);
+		if (if_ctx->roaming == true) {
+			if_ctx->roaming = false;
```

Comment:
This doesn't work for L3 roaming where DHCP is necessary, the proper fix would be for the application to implement NET_MGMT IPV4 events and then restart UDP throughput (zperf?).